frontend/src/views/Gallery.vue

```diff
@@ -29,6 +29,27 @@ const searchCursor = ref("");
 const emitter = inject("emitter");
 emitter.on("filter", onFilterChange);
 
+socket.on("scan:done", () => {
+  scanning.set(false);
+  emitter.emit("refreshGallery");
+  emitter.emit("snackbarShow", {
+    msg: "Scan completed successfully!",
+    icon: "mdi-check-bold",
+    color: "green",
+  });
+  socket.disconnect();
+});
+
+socket.on("scan:done_ko", (msg) => {
+  scanning.set(false);
+  emitter.emit("snackbarShow", {
+    msg: `Scan couldn't be completed. Something went wrong: ${msg}`,
+    icon: "mdi-close-circle",
+    color: "red",
+  });
+  socket.disconnect();
+});
```

Comment:
Moving these out of `scan()` so we're no longer initialing a new listener each time `scan()` is called